### PR TITLE
Competitive programming

### PR DESCRIPTION
**Pull request by akshat12000**
<hr>

## Description 📜

Given an array A consisting 0s, 1s and 2s. The task is to write a function that sorts the given array without using extra space and any sorting algorithm. #3058 

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [ ] C/#CPP
- [x] Competitive Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] Java#Script
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

<hr>

### CODE REVIEW

File: Competitive_Programming/C++/Array/Dutch National Flag Algorithm.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# Dutch National Flag Algorithm
+Given an array A[] consisting 0s, 1s and 2s. The task is to write a function that sorts the given array without using extra space and any sorting algorithm.
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Array/Dutch National Flag Algorithm.md:2]: add another header as `## Problem Statement`

**Review by ravireddy07 — CHANGES_REQUESTED** [review]
Great work @akshat12000, Just few changes required.. 

1. dont leave spaces in file name, use `underscore's` or `hyphens`.
2. checkout below image once, Those lines collided into single line. Please separate them.
3. Add a Title for Complexity.
Screenshot (411)

**Comment by akshat12000** [comment]
Respected sir,
I have resolved all the issues mentioned by you. Kindly review it once
again.

Best wishes
Akshat Bhatnagar

**Comment by ravireddy07** [comment]
well done @akshat12000

**Review by ravireddy07 — APPROVED** [review]

**Pull request merged**